AsteriskRuntimeFilter.php

```diff
@@ -26,64 +26,62 @@ public function __construct($config,$logger)
 
     public function __invoke($variables)
     {
-        // Get account index
-        $indexes = array();
-        foreach (array_keys($variables) as $key) {
-            if (strpos('account_username_', $key)) {
-                $indexes[] = str_replace('account_username_','',$key);
+        foreach (array_keys($variables) as $variable) {
+            if(substr($variable, 0, 18) != 'account_extension_') {
+                // Ignore all variables except those starting with "account_extension_"
+                continue;
+            }
+
+            $index = (integer) substr($variable, 18);
+            if($index <= 0) {
+                // index must be a positive integer
+                continue;
             }
-        }
-        if (empty($indexes)) $indexes[] = 1;
 
-        foreach ($indexes as $index) {
-            if (!empty($variables['mainextension_'.$index])) {
-                $extension = $variables['mainextension_'.$index];
-            } elseif (!empty($variables['extension_'.$index])) {
-                $extension = $variables['extension_'.$index];
-            } elseif (!empty($variables['account_username_'.$index])) {
-                $extension = $variables['account_username_'.$index];
-            } else {
-                return $variables;
+            // Initialize default values
+            $variables['account_call_waiting_' . $index] = '';
+            $variables['account_timeout_fwd_target_' . $index] = '';
+            $variables['account_busy_fwd_target_' . $index] = '';
+            $variables['account_always_fwd_target_' . $index] = '';
+            $variables['account_cftimeout_' . $index] = '';
+            $variables['account_dnd_enable_' . $index] = '';
+
+            $mainextension_match = array();
+            $extension = $variables[$variable];
+            if(preg_match("/^9\d(\d\d\d)$/", $extension, $mainextension_match)) {
```

Comment:
match main extensions with 2 or 4 digit
```suggestion
            if(preg_match("/^9\d(\d{2,4})$/", $extension, $mainextension_match)) {
```